consensus/consensus-types/src/block.rs

```diff
@@ -241,6 +244,17 @@ impl Block {
                 self.timestamp_usecs() > parent.timestamp_usecs(),
                 "Blocks must have strictly increasing timestamps"
             );
+
+            let current_ts = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Current time is before the UNIX_EPOCH!");
+
+            // we can say that too far is 5 minutes in the future
+            const TIMEBOUND: u64 = 300_000_000;
```

Comment:
I think we should be more cautious than that, my understanding is that depending on the configuration we can end up with rounds that will have a longer deadline. @zekun000 what would you suggest? I'm thinking that we could use the calculated max round deadline + 1sec instead of 5min

Reply:
we don't have access to the round deadline in the verify function

Reply:
yeah I was thinking about a `const fn` but actually IIUC the configuration for the deadline needs to be dynamically calculated

Reply:
any suggestion then? I feel like 5min is potentially too low but I might be wrong

Reply:
5mins seems reasonable to me, 5mins is almost forever :)

Reply:
or we can use 10 mins to align with the delta property (and bring it back to the spec)

Reply:
I'm going to ask you one more thing :D can you update the consensus specification at the right, as part of this PR, to indicate that we make this check now?